modules/mono/glue/cs_files/Quat.cs

```diff
@@ -73,65 +81,69 @@ public Quat CubicSlerp(Quat b, Quat preA, Quat postB, real_t t)
 
         public real_t Dot(Quat b)
         {
-            return x * b.x + y * b.y + z * b.z + w * b.w;
+            return X * b.X + Y * b.Y + Z * b.Z + W * b.W;
+        }
+
+        public Vector3 GetEuler()
```

Comment:
Here I've just used the GDScript implementation as a guide.

However I'm happy enough to deviate from that if people reach a consensus.

Reply:
Hmm, you're right that the GDScript and Core implementation works that way. However, this is inconsistent with the rest of Quat, and other Core math (ex: Vector3's only `get_`/`set_` method is for axis by index, and in this case "axis" is an unsuitable since it's an enum).

I may make a PR for the Core math to change Quat's Euler methods to be inline with the rest.

Reply:
Update: I have discovered that GDScript *does not allow function overloading*. However, we only have to have one of these named in GDScript (likely the setter). We could do `euler()` and `set_euler` in both and never have a `get_euler`. Or, we could overload `Set()` in C# instead of having `SetEuler()` for setting. But this might be an undesired inconsistency.

Reply:
> no function overloading

That explains why `Transform2d.xform()` takes a Variant parameter.  
http://docs.godotengine.org/en/latest/classes/class_transform2d.html#class-transform2d-xform

I would be leaning towards having a consistent interface. C# users reading the docs should be able follow along at home.